source/server/listener_manager_impl.cc

```diff
@@ -456,17 +456,23 @@ void ListenerImpl::setSocket(const Network::SocketSharedPtr& socket) {
 
 ListenerManagerImpl::ListenerManagerImpl(Instance& server,
                                          ListenerComponentFactory& listener_factory,
-                                         WorkerFactory& worker_factory)
-    : server_(server), factory_(listener_factory), stats_(generateStats(server.stats())),
+                                         WorkerFactory& worker_factory,
+                                         SystemTimeSource& system_time_source)
```

Comment:
Can `system_time_source` be dervied from the server `Instance` here?

Reply:
I don't think any of the structs in `Instance` have access. I could bundle it in this struct as we discussed doing in FactoryContext above?

Reply:
It seems like it would be something worth hanging of `Server::Instance`, it has server-wide scope.